modules/configuration/src/test/java/org/apache/ignite/internal/configuration/testframework/ConfigurationExtensionTest.java

```diff
@@ -50,4 +56,62 @@ public void injectConfiguration(
 
         assertEquals(300, paramCfg.joinTimeout().value());
     }
+
+    /** Tests that notifications work on injected configuration instance. */
+    @Test
+    public void notifications() {
+        List<String> log = new ArrayList<>();
+
+        fieldCfg.listen(ctx -> {
+            log.add("update");
```

Comment:
are listeners executed in the same thread?

Reply:
yes

Reply:
Not anymore, but hb rules still apply, collection doesn't have to be concurrent